cmds/rm/rm.go

```diff
@@ -33,6 +33,8 @@ var (
 		f bool
 	}
 	cmd = "rm [-Rrvif] file..."
+
+	input = bufio.NewReader(os.Stdin)
```

Comment:
Why is this global? Please move back to main().

Reply:
Oh, I see why.

Suggestion: 

Make `rm` take an io.Reader, wrap that in the bufio.NewReader.

Then, in the tests you can pass any io.Reader to `rm` you want.